### PR TITLE
Space agent no longer reset timeofdeath, support for ghostroles, players cannot steal aghosted xenos, ghost can take afk marines

### PR DESCRIPTION
**Pull request by BraveMole**
## About The Pull Request

Fixes: https://github.com/tgstation/TerraGov-Marine-Corps/issues/8143

you can no longer cheat respawn timer by spawning as a space agent

Support for ghost roles, aka it's possible to add ghost roles without killing respawn timer. Also allow admin memery without killing respawn timer

AFK marines are takable by ghosts, exactly like xenos.

## Why It's Good For The Game

QOL and bug fixes. AFK marines being free for the taking was approved by headmin

## Changelog
:cl:
fix: You can no longer steal aghosted bodies
fix: Spawning as a space agent no longer allow to bypass respawn timer
add: AFK marines can be taken by ghosts
/:cl:

### CODE REVIEW

**Comment by MereWalker** [comment]
I don't know of any admins cheating the respawn system but this is much liked since most of us tend to keep track of our time since becoming an agent resets the timer. Regardless this is a great idea and the other additions are great! Thank you!

**Comment by XSlayer300** [comment]
I accidentally did abuse this, once, tbh. I'm glad that ppl can't steal aghosted xenos anymore thankfully.

File: code/modules/mob/dead/observer/observer.dm [hunk 1]
```diff
@@ -368,6 +372,48 @@ GLOBAL_VAR_INIT(observer_default_invisibility, INVISIBILITY_OBSERVER)
 	mind.transfer_to(old_mob, TRUE)
 	return TRUE
 
+/mob/dead/observer/verb/take_ssd_mob()
+	set category = "Ghost"
+	set name = "Try to take SSD mob"
+
+	if(GLOB.key_to_time_of_death[key] + AFK_TIMER < world.time)
+		to_chat(src, span_warning("You died too recently to be able to take a new mob"))
```
Comment (XSlayer300) [inline code/modules/mob/dead/observer/observer.dm:380]:
```suggestion
		to_chat(src, span_warning("You died too recently to be able to take a new mob."))
```

**Comment by XSlayer300** [comment]
I wonder, how long will be the AFK timer for the marines be? 5 minutes I guess?

**Comment by BraveMole** [comment]
> I wonder, how long will be the AFK timer for the marines be? 5 minutes I guess?

5 minutes, but that's up for discussion

File: tgmc.dme [hunk 2]
```diff
@@ -336,6 +336,7 @@
 #include "code\datums\elements\_element.dm"
 #include "code\datums\elements\accelerate_on_cross.dm"
 #include "code\datums\elements\attachment.dm"
+#include "code\datums\elements\accelerate_on_cross.dm"
```
Comment (BraveMole) [inline tgmc.dme:339]:
```suggestion
```

File: code/modules/mob/dead/observer/observer.dm [hunk 3]
```diff
@@ -373,6 +377,48 @@ GLOBAL_VAR_INIT(observer_default_invisibility, INVISIBILITY_OBSERVER)
 	mind.transfer_to(old_mob, TRUE)
 	return TRUE
 
+/mob/dead/observer/verb/take_ssd_mob()
+	set category = "Ghost"
+	set name = "Try to take SSD mob"
+
+	if(GLOB.key_to_time_of_death[key] + AFK_TIMER < world.time)
+		to_chat(src, span_warning("You died too recently to be able to take a new mob."))
```
Comment (TiviPlus) [inline code/modules/mob/dead/observer/observer.dm:385]: ...afk timer?

File: code/modules/mob/living/living.dm [hunk 4]
```diff
@@ -839,3 +837,49 @@ below 100 is not dizzy
...
+		if(MOB_CONNECTED, MOB_DISCONNECTED)
+			if(afk_timer_id)
+				deltimer(afk_timer_id)
+				afk_timer_id = null
+		if(MOB_RECENTLY_DISCONNECTED)
+			if(afk_status == MOB_RECENTLY_DISCONNECTED)
+				if(timeleft(afk_timer_id) > afk_timer)
+					deltimer(afk_timer_id) //We'll go with the shorter timer.
+				else
+					return
```
Comment (TiviPlus) [inline code/modules/mob/living/living.dm:853]: early return

**Review by TiviPlus — CHANGES_REQUESTED** [review]

**Review by TiviPlus — APPROVED** [review]

**Pull request merged**